homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -103,52 +147,75 @@ class Hub(SurePetcareBinarySensor):
 
     def __init__(self, _id: int, spc: SurePetcareAPI) -> None:
         """Initialize a Sure Petcare Hub."""
-        super().__init__(_id, spc, DEVICE_CLASS_CONNECTIVITY)
+        super().__init__(_id, spc, DEVICE_CLASS_CONNECTIVITY)  # , EntityType.HUB)
 
-    @callback
-    def _async_update(self) -> None:
-        """Get the latest data and update the state."""
-        surepy_entity = self._spc.states[self._id]
-        state = surepy_entity.raw_data()["status"]
-        self._attr_is_on = self._attr_available = bool(state["online"])
-        if surepy_entity.raw_data():
-            self._attr_extra_state_attributes = {
-                "led_mode": int(surepy_entity.raw_data()["status"]["led_mode"]),
+    @property
+    def available(self) -> bool:
+        """Return true if entity is available."""
+        return bool(self._state["online"])
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if entity is online."""
+        return self.available
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return the state attributes of the device."""
+        attributes = None
+        if self._surepy_entity.raw_data():
+            attributes = {
+                "led_mode": int(self._surepy_entity.raw_data()["status"]["led_mode"]),
                 "pairing_mode": bool(
```

Comment:
This is all reverting work that is being done on the core repository and are unneeded.